### PR TITLE
Non-allocating iterators for flash IntMap

### PR DESCRIPTION
**Pull request by SmerkyG**
Fix for issue #3762

### CODE REVIEW

File: std/flash/_std/haxe/ds/IntMap.hx [hunk 1]
```diff
@@ -74,5 +90,65 @@ package haxe.ds;
...
+#if !as3
+
+// this version uses __has_next__/__forin__ special SWF opcodes for iteration with no allocation
+
+@:allow(haxe.ds.IntMap)
+private class IntMapKeysIterator {
+	var h:flash.utils.Dictionary;
+	var index : Int;
+	var nextIndex : Int;
+	var isReserved : Bool;
```
Comment (nadako) [inline std/flash/_std/haxe/ds/IntMap.hx]: this one is rudimentary

**Comment by SmerkyG** [comment]
Sorry, there's some issue travis is noticing in as3 that I don't understand but seems unrelated:
```
/home/travis/build/SmerkyG/haxe/tests/unit/bin/as3/haxe/xml/Parser.as(21): col: 62 Error: Required parameters are not permitted after optional parameters.
		static protected function doParse(str : String,p : int = 0,strict : Boolean,parent : Xml = null) : int {
		                                                           ^
```

File: std/flash/_std/haxe/ds/IntMap.hx (hunk 1 above)
Comment (SmerkyG) [inline std/flash/_std/haxe/ds/IntMap.hx]: Oops, good call. Got rid of that.

**Comment by Simn** [comment]
Could you sync again, the unit tests should work now.

**Comment by SmerkyG** [comment]
I added a non-allocating iterator for UnsafeStringMap to this as well, since that one was also allocating. (I actually came across the issue when profiling real code!)

**Comment by Simn** [comment]
@ncannasse: Please check if you want this merged for 3.3.

**Pull request merged**